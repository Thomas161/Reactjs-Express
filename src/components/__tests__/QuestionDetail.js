import {mapStateToProps, QuestionDetailDisplay} from '../QuestionDetail';
import renderer from 'react-test-renderer';
import React from 'react';

describe(`The Question Component`, ()=> {
    describe(`The Container Element`, ()=> {
        describe(`mapStateToProps`, ()=>{
            it("should map state to props correctly", ()=> {
                const sampleQuestion = {
question_id:13,
body:"Space is big"
                };
                
                const app = {
                    questions: [sampleQuestion]
                };
                const ownProps = {
                    question_id: 13
                };
                const componentState = mapStateToProps(app,ownProps);
                console.log(componentState);
                expect(componentState).toEqual(sampleQuestion);
            });
        });
    });
    describe('The display element', ()=> {
        it(`Should not regress`, ()=> {
            const tree = renderer.create(
                <QuestionDetailDisplay 
                title="The meaning of life"
                body="15"
                answer_count={0}
                tags={['Hitchhiker']}
                />
            );

            expect(tree.toJSON()).toMatchSnapshot();
    });


    //lack of assertion is a pass
});
});