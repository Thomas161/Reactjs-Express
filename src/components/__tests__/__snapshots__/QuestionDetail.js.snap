// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`The Question Component The display element Should not regress 1`] = `
<div>
  <h3
    className="mb-2"
  >
    The meaning of life
  </h3>
  <div>
    <div
      className="mb-3"
    >
      <div>
        <code>
          Hitchhiker
        </code>
      </div>
    </div>
    <div>
      <p>
        15
      </p>
    </div>
    <div>
      0
       Answers
    </div>
  </div>
</div>
`;
